dhall/src/Dhall.hs

```diff
@@ -211,11 +237,36 @@ type MonadicExtractor s a = Either (ExtractErrors s a)
 -}
 typeError :: Expr s a -> Expr s a -> Extractor s a b
 typeError expected actual =
-    Failure . ExtractErrors . pure . TypeMismatch $ InvalidDecoder expected actual
+    Failure . DhallErrors . pure . TypeMismatch $ InvalidDecoder expected actual
+
+{-| Like `typeError`, but the expected type is not yet determined.
+-}
+typeError' :: Expector (Expr s a) -> Expr s a -> Extractor s a b
```

Comment:
Perhaps rename this to `expectorError`?  I try to avoid using name containing a  `'` in the public API

Reply:
Since I changed `typeError'` to `typeError` there is no `'` in the public API anymore.
I'd like to keep the name if it's ok.

Reply:
Yeah, I prefer this approach of changing the type of `typeError`